api/application.go

```diff
@@ -49,12 +49,13 @@ func (h ApplicationHandler) AddRoutes(e *gin.Engine) {
 	routeGroup.DELETE(ApplicationsRoot, h.DeleteList)
 	routeGroup.DELETE(ApplicationRoot, h.Delete)
 	// Tags
+	routeGroup = e.Group("/")
 	routeGroup.Use(auth.Required("applications"))
```

Comment:
**hint**: As a result of the missing:
```
routeGroup = e.Group("/")
```
The 
```
auth.Required("applications")
```
Added the `Required` AFTER the `Transaction`.

Reply:
Great job finding that, what a nightmare to debug. What do you think about maybe adding a safeguard to the top of the Transaction handler to catch situations like this? Gin exposes the names of functions in the handler chain to us, we could check to see that the Transaction handler is second to last and abort if it isn't to avoid a deadlock:

```go
func Transaction(ctx *gin.Context) {
	names := ctx.HandlerNames()
	parts := strings.Split(names[len(names) - 2], "/")
	if parts[len(parts) - 1] != "api.Transaction" {
		_ = ctx.Error(liberr.New("invalid handler chain", "chain", names))
		ctx.Abort()
		return
	}
...
```